### PR TITLE
Add like predicate to Show Functions query

### PR DESCRIPTION
**Pull request by Praveen2112**

### CODE REVIEW

File: presto-docs/src/main/sphinx/sql/show-functions.rst [hunk 1]
```diff
@@ -7,9 +7,10 @@ Synopsis
 
 .. code-block:: none
 
-    SHOW FUNCTIONS
+    SHOW FUNCTIONS [ LIKE pattern ]
```
Comment (findepi) [inline presto-docs/src/main/sphinx/sql/show-functions.rst]: ESCAPE?

File: presto-tests/src/main/java/io/prestosql/tests/AbstractTestQueries.java [hunk 2]
```diff
@@ -2599,6 +2599,35 @@ public void testShowFunctions()
         assertFalse(functions.containsKey("like"), "Expected function names " + functions + " not to contain 'like'");
     }
 
+    @Test
+    public void testShowFunctionsLike()
+    {
+        MaterializedResult result = computeActual("SHOW FUNCTIONS LIKE 'split%'");
+        ImmutableMultimap<String, MaterializedRow> functions = Multimaps.index(result.getMaterializedRows(), input -> {
```
[the remark this answers is not in the document]
Comment (martint) [reply presto-tests/src/main/java/io/prestosql/tests/AbstractTestQueries.java]: Also, move this to an engine-specific test like `io.prestosql.sql.query.TestXXX`. You can use this shape, instead:

```java
assertions.assertQuery(
    "SHOW FUNCTIONS LIKE 'split%'", 
    "VALUES (..., ..., ...)");
```

File: presto-tests/src/main/java/io/prestosql/tests/AbstractTestQueries.java [hunk 3]
```diff
@@ -2599,6 +2599,35 @@ public void testShowFunctions()
...
+        assertEquals(functions.size(), 5);
+        assertEquals(functions.get("split_part").asList().get(0).getField(1), "varchar(x)");
+        assertEquals(functions.get("split_part").asList().get(0).getField(2), "varchar(x), varchar(y), bigint");
+        assertEquals(functions.get("split_part").asList().get(0).getField(3), "scalar");
+    }
+
+    @Test
+    public void testShowFunctionsLikeWithEscape()
+    {
+        MaterializedResult result = computeActual("SHOW FUNCTIONS LIKE 'split$_to$_%' ESCAPE '$'");
```
Comment (martint) [inline presto-tests/src/main/java/io/prestosql/tests/AbstractTestQueries.java]: Same as above.

**Comment by martint** [comment]
Travis failure is related:

```
[ERROR]   Run 50: TestSqlParserErrorHandling.testStatement:175 Error message mismatch for query:

SHOW FUNCTION LIKE '%$_%' ESCAPE

 expected [line 1:32: mismatched input '<EOF>'. Expecting: <string>] but found [line 1:6: mismatched input 'FUNCTION'. Expecting: 'CATALOGS', 'COLUMNS', 'CREATE', 'CURRENT', 'FUNCTIONS', 'GRANTS', 'ROLE', 'ROLES', 'SCHEMAS', 'SESSION', 'STATS', 'TABLES']
```

File: presto-parser/src/test/java/io/prestosql/sql/parser/TestSqlParserErrorHandling.java [hunk 4]
```diff
@@ -141,7 +141,9 @@
                                 "'GROUP', 'HAVING', 'INTERSECT', 'LIMIT', 'OFFSET', 'OR', 'ORDER', 'SELECT', 'TABLE', 'UNION', 'VALUES', 'WHERE', '[', '||', <EOF>, " +
                                 "<identifier>, <predicate>"},
                 {"SHOW CATALOGS LIKE '%$_%' ESCAPE",
-                        "line 1:33: mismatched input '<EOF>'. Expecting: <string>"}
+                        "line 1:33: mismatched input '<EOF>'. Expecting: <string>"},
+                {"SHOW FUNCTION LIKE '%$_%' ESCAPE",
```
Comment (martint) [inline presto-parser/src/test/java/io/prestosql/sql/parser/TestSqlParserErrorHandling.java]: This should be `SHOW FUNCTIONS LIKE ...`

**Comment by findepi** [comment]
That's so useful. I use it very often recently. Thanks!

**Pull request merged**